pyomo/contrib/doe/DoE_package.py

```diff
@@ -0,0 +1,1812 @@
+
+                # solve model
+                if self.mode =='sequential_sipopt':
+                    m_sipopt = sensitivity_calculation('sipopt', mod, list_original, list_perturb, tee=self.tee_opt, solver_options='ma57')
+                # TODO: add k_aug solver
+                else:
+                    m_sipopt = sensitivity_calculation('k_aug', mod, list_original, list_perturb, tee=True)
+
+                # extract sipopt result
+                for j in self.measurement_variables:
+                    for t in self.measurement_timeset:
+                        # fetch the measurement variable
+                        measure_var = getattr(m_sipopt,j)
+                        # check if this variable is fixed
+                        if (measure_var[0,t].fixed == True):
+                            perturb_value = value(measure_var[0,t])
+                            #if self.verbose:
+                            #    print(measure_var[0, t], ' is fixed')
+
+                        else:
+                            # if it is not fixed, record its perturbed value
+                            perturb_value = eval('m_sipopt.sens_sol_state_1[m_sipopt.' + j + '[0,' + str(t) + ']]')
+                        perturb_mea.append(perturb_value)
+
+                        # base case values
+                        base_value = eval('m_sipopt.'+j+'[0,' + str(t) + '].value')
+                        base_mea.append(base_value)
+                # store extracted measurements
+                all_perturb_measure.append(perturb_mea)
+                all_base_measure.append(base_mea)
+
+
+            # After collecting outputs from all scenarios, calculate sensitivity
+            for count, para in enumerate(self.param):
+                list_jac = []
+                for i in range(len(all_perturb_measure[0])):
+                    if self.scale_nominal_param_value:
+                        sensi = -(all_perturb_measure[count][i] - all_base_measure[count][i]) / self.step * self.scale_constant_value
+                    else:
+                        sensi = -(all_perturb_measure[count][i] - all_base_measure[count][i]) / self.step /self.param_init[para] * self.scale_constant_value
+                    list_jac.append(sensi)
+                # get Jacobian dict, keys are parameter name, values are sensitivity info
+                jac[para] = list_jac
+
+            # analyze results
+            if specified_prior is None:
+                prior_in_use = self.prior_FIM
+            else:
+                prior_in_use = specified_prior
+
+            # analyze results
+            FIM_analysis = FIM_result(self.param, prior_FIM=prior_in_use, store_FIM=FIM_store_name, scale_constant_value=self.scale_constant_value)
+
+            self.jac = jac
+            #self.square_result = None
+
+            return FIM_analysis
+
+
+    def sequential_exp(self, design_values_set, mode='sequential_finite', tee_option=False,
```

Comment:
Should we instead call this function "simulate_sequential_experiments" or something similar?

Reply:
Or "generate_sequential_experiments"

Reply:
What is the user wanted to sequentially optimize the experiments?